hyperparameter_train_network.py

```diff
@@ -41,69 +46,144 @@ def color_cat(s):
         return 2
     elif s == 'RED':
         return 3
-    else:
-        return 4
+    raise RuntimeError('Unrecognized color:', s)
 
 ## turn all the words into numbers, replace with more elgeant solution for varying amounts of letters and words later
 def encode_state(state):
-    state.block0_side1_letter = state.block0_side1_letter.apply(ord)-ord('A')+1
-    state.block0_side2_letter = state.block0_side2_letter.apply(ord)-ord('A')+1
-    state.block1_side1_letter = state.block1_side1_letter.apply(ord)-ord('A')+1
-    state.block1_side2_letter = state.block1_side2_letter.apply(ord)-ord('A')+1
-    state.block0_side1_color=state.block0_side1_color.apply(color_cat)
-    state.block0_side2_color=state.block0_side2_color.apply(color_cat)
-    state.block1_side1_color=state.block1_side1_color.apply(color_cat)
-    state.block1_side2_color=state.block1_side2_color.apply(color_cat)
-    return state;  
-# take in the instruction from the commander, i went with 100 max length just as a guess
-words_input = Input(shape=(MAX_LEN,N_SYMBOLS), name='words_input')
-
-# # This embedding layer will encode the input sequence
-# # into a sequence of dense 512-dimensional vectors.
-# x1 = Embedding(output_dim=512, input_dim=10000, input_length=100)(words_input)
-words_lstm=LSTM(32)(words_input)
-
-#the point to pass in
-position_input = Input(shape=(2,), name='position_input')
-
-#the state, max bblocks is set above
-state_input = Input(shape=(14,),name='state_input')
-#state_input1 = keras.layers.Flatten()(state_input)
-network_input = keras.layers.concatenate([words_lstm, position_input,state_input])
-
-# # do actual building here, pull out later functionally do hyper parameter building
-Nodes=[3,6,9]
-network= BuildNet(Nodes,network_input);
-
-block_id_output = Dense(1, activation='relu',name='block_id_output')(network)
-point_output= Dense(2,activation='relu',name='point_output')(network)
-
-
-final_network=Model(inputs=[state_input,words_input,position_input],outputs=[block_id_output,point_output])
-final_network.compile(loss='mean_squared_error',
-        optimizer='Adam',
-        metrics=['accuracy']) 
+    for i in range(len(state)):
```

Comment:
Is there a reason you're iterating through indices rather than just `for curr_state in state:`?

Reply:
No